### PR TITLE
Remove XML configuration option

### PR DESCRIPTION
**Pull request by kbaley**
Connects to https://github.com/Particular/PlatformDevelopment/issues/729

### CODE REVIEW

File: src/NServiceBus.Persistence.AzureStorage.ComponentTests/Persisters/When_using_AzureStorageSagaGuard.cs [hunk 1]
```diff
@@ -17,8 +16,7 @@ public void Should_not_allow_invalid_connection_string(string connectionString)
         [Test]
         public void Should_validate_all_default_settings_for_a_new_config()
         {
-            var config = new AzureSagaPersisterConfig();
-            Assert.AreEqual(AzureStorageSagaDefaults.CreateSchema, config.CreateSchema);
+            Assert.AreEqual(AzureStorageSagaDefaults.CreateSchema, AzureStorageSagaDefaults.CreateSchema);
```
Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage.ComponentTests/Persisters/When_using_AzureStorageSagaGuard.cs]: As far as I see it compares the default with itself. Shouldn't the test be removed then?

File: src/NServiceBus.Persistence.AzureStorage.ComponentTests/Persisters/When_using_AzureSubscriptionStorageGuard.cs [hunk 2]
```diff
@@ -33,9 +32,8 @@ public void Should_not_allow_invalid_table_name(string tableName)
         [Test]
         public void Should_validate_all_default_settings_for_a_new_config()
         {
-            var config = new AzureSubscriptionStorageConfig();
-            Assert.AreEqual(AzureSubscriptionStorageDefaults.CreateSchema, config.CreateSchema);
-            Assert.AreEqual(AzureSubscriptionStorageDefaults.TableName, config.TableName);
+            Assert.AreEqual(AzureSubscriptionStorageDefaults.CreateSchema, AzureSubscriptionStorageDefaults.CreateSchema);
```
Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage.ComponentTests/Persisters/When_using_AzureSubscriptionStorageGuard.cs]: Again, comparing defaults. It looks like this test can be removed.

File: src/NServiceBus.Persistence.AzureStorage.ComponentTests/Subscriptions/SuscriptionTestHelper.cs [hunk 3]
```diff
@@ -3,29 +3,27 @@
...
     public class SubscriptionTestHelper
     {
         internal static ISubscriptionStorage CreateAzureSubscriptionStorage()
         {
-            var config = new AzureSubscriptionStorageConfig();
             var connectionString = AzurePersistenceTests.GetConnectionString();
             var account = CloudStorageAccount.Parse(connectionString);
 
-            var table = account.CreateCloudTableClient().GetTableReference(config.TableName);
+            var table = account.CreateCloudTableClient().GetTableReference(AzureSubscriptionStorageDefaults.TableName);
```
Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage.ComponentTests/Subscriptions/SuscriptionTestHelper.cs]: Is there a way of overriding the default? How the previous version configured with a custom config, can provide the same value when migrating?

File: src/NServiceBus.Persistence.AzureStorage/Config/AzureSagaPersisterConfig.cs [hunk 4]
```diff
@@ -1,52 +0,0 @@
-namespace NServiceBus.Config
-{
-    using System.Configuration;
-
-    /// <summary>
-    /// Config section for the Azure Saga Persister
-    /// </summary>
-    public class AzureSagaPersisterConfig:ConfigurationSection
```
Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage/Config/AzureSagaPersisterConfig.cs]: Would you consider providing an `[ObsoloteEx]` obsoleting the value and giving a meaningful build error?

File: src/NServiceBus.Persistence.AzureStorage/Config/AzureSubscriptionStorageConfig.cs [hunk 5]
```diff
@@ -1,59 +0,0 @@
-namespace NServiceBus.Config
-{
-    using System.Configuration;
-    using Subscriptions;
-
-    public class AzureSubscriptionStorageConfig : ConfigurationSection
```
Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage/Config/AzureSubscriptionStorageConfig.cs]: Again, would `[ObsoleteEx]` make sense in here?

File: src/NServiceBus.Persistence.AzureStorage/Config/AzureTimeoutPersisterConfig.cs [hunk 6]
```diff
@@ -1,76 +0,0 @@
-namespace NServiceBus.Config
-{
-    using System.Configuration;
-    using Timeout;
-
-    public class AzureTimeoutPersisterConfig : ConfigurationSection
```
Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage/Config/AzureTimeoutPersisterConfig.cs]: `[ObsoleteEx]` ?

File: src/NServiceBus.Persistence.AzureStorage/Timeout/AzureStorageTimeoutPersistence.cs [hunk 7]
```diff
@@ -14,14 +14,14 @@ internal AzureStorageTimeoutPersistence()
...
-                var config = s.GetConfigSection<AzureTimeoutPersisterConfig>() ?? new AzureTimeoutPersisterConfig();
-                s.SetDefault("AzureTimeoutStorage.ConnectionString", config.ConnectionString);
-                s.SetDefault("AzureTimeoutStorage.CreateSchema", config.CreateSchema);
-                s.SetDefault("AzureTimeoutStorage.TimeoutManagerDataTableName", config.TimeoutManagerDataTableName);
-                s.SetDefault("AzureTimeoutStorage.TimeoutDataTableName", config.TimeoutDataTableName);
-                s.SetDefault("AzureTimeoutStorage.CatchUpInterval", config.CatchUpInterval);
-                s.SetDefault("AzureTimeoutStorage.PartitionKeyScope", config.PartitionKeyScope);
-                s.SetDefault("AzureTimeoutStorage.TimeoutStateContainerName", config.TimeoutStateContainerName);
+                var defaultConnectionString = ConfigurationManager.AppSettings["NServiceBus/Persistence"];
+                s.SetDefault("AzureTimeoutStorage.ConnectionString", defaultConnectionString);
```
Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage/Timeout/AzureStorageTimeoutPersistence.cs]: Would you consider providing `consts` for this configuration keys?

File: src/NServiceBus.Persistence.AzureStorage/Config/AzureSagaPersisterConfig.cs (hunk 4 above)
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureStorage/Config/AzureSagaPersisterConfig.cs]: I'm bearing in mind that ASP (or P.AS) is a brand new package.
Saying that, it's the same "candy" in a new wrapper. So perhaps for better guidance obsoletes would help.

Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage/Config/AzureSagaPersisterConfig.cs]: Thx for reminding me of this @SeanFeldman . Although I've been seeing v1 in docs it slipped my mind when I was writing the comment above. I agree with you though, that users of the previous package may just want to copy the configuration over.

File: src/NServiceBus.Persistence.AzureStorage.ComponentTests/Subscriptions/SuscriptionTestHelper.cs (hunk 3 above)
Comment (kbaley) [inline src/NServiceBus.Persistence.AzureStorage.ComponentTests/Subscriptions/SuscriptionTestHelper.cs]: We have code config already in place via extension methods (e.g. `ConfigureAzureTimeoutStorage`). It's new to V1 and wasn't in NSB.Azure.

File: src/NServiceBus.Persistence.AzureStorage/Config/AzureSagaPersisterConfig.cs (hunk 4 above)
Comment (kbaley) [inline src/NServiceBus.Persistence.AzureStorage/Config/AzureSagaPersisterConfig.cs]: I should have verified but I assumed obsoleting a config section wouldn't generate a build error.

Comment (SimonCropp) [inline src/NServiceBus.Persistence.AzureStorage/Config/AzureSagaPersisterConfig.cs]: Config sections can be used in code via the iprovideconfiguration api

**Comment by SimonCropp** [comment]
@dbelcham did u forget to delete the branch?

**Pull request merged**